amazon-msk/cdk/lib/IotIngestAndControlStack.ts

```diff
@@ -415,43 +413,46 @@ export class IotIngestAndControlStack extends cdk.Stack {
 
     autoScalingGroup.attachToNetworkTargetGroup(targetGroup);
 
-    const metricsNamespace = context.cloudwatch?.metrics?.namespace ?? 'zilla';
+    if (context.cloudwatch?.metrics) {
+      const metricsNamespace = context.cloudwatch.metrics.namespace;
 
-    const metricWorkerUtilization = new cw.Metric({
-      namespace: metricsNamespace,
-      metricName: 'engine.worker.utilization',
-      statistic: 'Average',
-      period: cdk.Duration.minutes(5),
-    });
+      const metricWorkerUtilization = new cw.Metric({
+        namespace: metricsNamespace,
+        metricName: 'engine.worker.utilization',
+        statistic: 'Average',
+        period: cdk.Duration.minutes(5),
+      });
 
-    const metricWorkerCount = new cw.Metric({
-      namespace: metricsNamespace,
-      metricName: 'engine.worker.count',
-      statistic: 'Average',
-      period: cdk.Duration.minutes(5),
-    });
+      const metricWorkerCount = new cw.Metric({
+        namespace: metricsNamespace,
+        metricName: 'engine.worker.count',
+        statistic: 'Average',
+        period: cdk.Duration.minutes(5),
+      });
 
-    const metricOverallWorkerUtilization = new cw.MathExpression({
-      label: 'OverallWorkerUtilization',
-      expression: 'm1 / m2',
-      usingMetrics: {
-        m1: metricWorkerUtilization,
-        m2: metricWorkerCount,
-      },
-    });
+      const metricOverallWorkerUtilization = new cw.MathExpression({
+        label: 'OverallWorkerUtilization',
+        expression: 'm1 / m2',
+        usingMetrics: {
+          m1: metricWorkerUtilization,
+          m2: metricWorkerCount,
+        },
+      });
 
-    const scalingSteps = context.scalingSteps ??
-    [
-      { upper: 0.30, change: -1 },
-      { lower: 0.80, change: +2 }
-    ]
-
-    autoScalingGroup.scaleOnMetric('WorkerUtilizationStepScaling', {
-      metric: metricOverallWorkerUtilization,
-      adjustmentType: autoscaling.AdjustmentType.CHANGE_IN_CAPACITY,
-      scalingSteps,
-      estimatedInstanceWarmup: cdk.Duration.minutes(2),
-    });
+      const scalingSteps = context.scalingSteps ??
+        [
+          { upper: 0.30, change: -1 },
+          { lower: 0.80, change: +2 }
+        ]
+
+      autoScalingGroup.scaleOnMetric('WorkerUtilizationStepScaling', {
+        metric: metricOverallWorkerUtilization,
+        adjustmentType: autoscaling.AdjustmentType.CHANGE_IN_CAPACITY,
+        scalingSteps,
+        estimatedInstanceWarmup: cdk.Duration.minutes(2),
+        cooldown: cdk.Duration.minutes(3)
```

Comment:
How do we determine the `warmup` and `cooldown` durations?

Does it make sense to make these configurable also, given that different `zilla.yaml` configurations might need longer to warmup for example?